ui/src/main/js/application/auth/SamlForm.js

```diff
@@ -0,0 +1,358 @@
+import ConcreteConfigurationForm from 'common/configuration/global/concrete/ConcreteConfigurationForm';
+
+import CheckboxInput from 'common/component/input/CheckboxInput';
+import LabeledField from 'common/component/input/field/LabeledField';
+import RadioInput from 'common/component/input/RadioInput';
+import TextInput from 'common/component/input/TextInput';
+import GeneralButton from 'common/component/button/GeneralButton';
+import UploadFileButtonField from 'common/component/input/field/UploadFileButtonField';
+
+import * as ConfigurationRequestBuilder from 'common/util/configurationRequestBuilder';
+import * as FieldModelUtilities from 'common/util/fieldModelUtilities';
+import * as HttpErrorUtilities from 'common/util/httpErrorUtilities';
+
+import BlackDuckSSOConfigImportModal from './BlackDuckSSOConfigImportModal';
+
+const radioOptions = [{
+    name: 'url',
+    value: 'URL',
+    label: 'URL'
+}, {
+    name: 'xml',
+    value: 'FILE',
+    label: 'XML File'
+}];
+
+const useStyles = createUseStyles({
+    samlForm: {
+        padding: [0, '20px']
+    },
+    fillForm: {
+        padding: '0.5rem',
+        display: 'inline-flex'
+    }
+});
+
+const SamlForm = ({ csrfToken, errorHandler, readonly, fileDelete, fileRead, fileWrite }) => {
+    const classes = useStyles();
+    const [formData, setFormData] = useState({});
+    const [errors, setErrors] = useState(HttpErrorUtilities.createEmptyErrorObject());
+    const [showBlackDuckSSOImportModal, setShowBlackDuckSSOImportModal] = useState(false);
+    const samlRequestUrl = `${ConfigurationRequestBuilder.AUTHENTICATION_SAML_API_URL}`;
+
+    const importBlackDuckSSOConfigLabel = 'Retrieve Black Duck SAML Configuration';
+    const importBlackDuckSSOConfigDescription = 'Fills in some of the form fields based on the SAML configuration from the chosen Black Duck server (if a SAML configuration exists).';
+
+    const fetchData = async () => {
+        const response = await ConfigurationRequestBuilder.createReadRequest(samlRequestUrl, csrfToken);
+        const data = await response.json();
+        if (data) {
+            setFormData(data);
+        }
+
+        if (data.status === 404) {
+            setFormData({ ...data, metadataMode: 'URL' });
+        }
+    };
+
+    function updateData() {
+        return ConfigurationRequestBuilder.createUpdateRequest(samlRequestUrl, csrfToken, undefined, formData);
+    }
+
+    function deleteData() {
+        return ConfigurationRequestBuilder.createDeleteRequest(samlRequestUrl, csrfToken);
+    }
+
+    function postData() {
+        return ConfigurationRequestBuilder.createNewConfigurationRequest(samlRequestUrl, csrfToken, formData);
+    }
+
+    function handleValidation() {
+        if (formData.status === 404) {
+            delete formData.status;
+            delete formData.message;
+            delete formData.error;
+            delete formData.path;
+        }
+
+        // // HACKY Remove after testing on Michaels branch
```

Comment:
We should resolve this TODO if we can

Reply:
Thanks! Removed that and cleaned up this portion of the code 👍